src/button-with-icon/button-with-icon.style.tsx

```diff
@@ -0,0 +1,172 @@
+import styled, { css } from "styled-components";
+import { Color } from "../color/color";
+import { MediaQuery } from "../media/media";
+import { ComponentLoadingSpinner } from "../shared/component-loading-spinner/component-loading-spinner";
+import { TextStyleHelper } from "../text";
+import { ButtonIconPosition, MainButtonSize, MainButtonStyle } from "./types";
+import { DesignToken } from "../design-token";
+
+// =============================================================================
+// STYLE INTERFACES
+// =============================================================================
+export interface MainStyleProps {
+    $buttonStyle: MainButtonStyle;
+    $buttonIcon: JSX.Element;
+    $buttonIconPosition?: ButtonIconPosition | undefined;
+    $buttonSizeStyle?: MainButtonSize | undefined;
+    $buttonIsDanger?: boolean;
+}
+
+// =============================================================================
+// STYLING
+// =============================================================================
+export const Main = styled.button<MainStyleProps>`
```

Comment:
As per previous discussion, to reuse `Main` in `src/button` and add custom styling if necessary instead of copying.